component/azstorage/block_blob.go

```diff
@@ -582,11 +583,21 @@ func (bb *BlockBlob) List(prefix string, marker *string, count int32) ([]*intern
 				return blobList, nil, err
 			}
 		} else {
+			var mode os.FileMode
+			if blobInfo.Properties.Permissions != nil {
+				mode, err = getFileMode(*blobInfo.Properties.Permissions)
+				if err != nil {
+					log.Err("BlockBlob::List : Failed to get file mode for %s [%s]", *blobInfo.Name, err.Error())
+					return blobList, nil, err
+				}
+			} else {
```

Comment:
The else statement can be removed by initializing the mode value to 0 at line: 586
var mode os.FileMode = 0 // Default initialization to 0